pkg/trino/driver/driver.go

```diff
@@ -17,14 +18,28 @@ const DriverName string = "trino"
 func Open(settings models.TrinoDatasourceSettings) (*sql.DB, error) {
 	skipVerify := false
 	sslCert := ""
+	var clientCert tls.Certificate
 	if settings.Opts.TLS != nil {
 		skipVerify = settings.Opts.TLS.InsecureSkipVerify
 		sslCert = settings.Opts.TLS.CACertificate
 	}
+	if settings.Opts.TLS.ClientCertificate != "" {
+		if settings.Opts.TLS.ClientKey == "" {
+			return nil, errors.New("client certificate was configured without a client key")
+		}
+		var err error
+		clientCert, err = tls.X509KeyPair(
+			[]byte(settings.Opts.TLS.ClientCertificate),
+			[]byte(settings.Opts.TLS.ClientKey))
+		if err != nil {
+			return nil, errors.Join(errors.New("failed to load client certificate"), err)
+		}
+	}
 	client := &http.Client{
 		Transport: &http.Transport{
 			TLSClientConfig: &tls.Config{
 				InsecureSkipVerify: skipVerify,
+				Certificates:       []tls.Certificate{clientCert},
```

Comment:
Is it ok to create a slice with a nil certificate? Maybe you should append to a slice after calling `tls.X509KeyPair()`.

Reply:
I can do that. however, from my manual testing agains our Trino server it didn't seem to cause any issues. I tested with just basic auth as well as just TLS client auth.